charts/cryostat/templates/auth2_proxy.yaml

```diff
@@ -3,6 +3,4 @@ kind: Secret
 metadata:
   name: {{ .Release.Name }}-auth-proxy-secret
 data:
-  # auth_proxy_htpasswd: {{ .Files.Get ".oauth2proxy.htpasswdFile" | b64enc | quote }}
-  # FIXME this is a hack, the user definition should be pulled out of values.yaml: .authentication.basicAuth
-  auth_proxy_htpasswd: {{ "user:$2y$05$.p1/68oBWKX1FhAtZOArYOoNoMqsB4yuUNXGOeSATHPq3geKqEaZS" | b64enc | quote }}
+    auth_proxy_htpasswd: {{ .Values.authentication.basicAuth.secretName | b64enc | quote }}
```

Comment:
This means that the helm chart will create a new Secret named ex. `cryostat-auth-proxy-secret`, containing the data `auth_proxy_htpasswd: another-secret-name`. That doesn't seem quite right - we don't want/need to have this second level of indirection with a Secret pointing to another Secret.

Instead, the user's provided Secret should be mounted as a volume, using the `basicAuth.secretName` and `basicAuth.filename` they provide.

https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod

In `deployment.yaml` we already have parts of that set up:

```yaml
      {{- if .Values.authentication.basicAuth.enabled }}
      - name: basic-auth-secret
        secret:
          secretName: {{ .Values.authentication.basicAuth.secretName }}
      {{- end }}
```

```yaml
            {{- if .Values.authentication.basicAuth.enabled }}
            - name: basic-auth-secret
              mountPath: /etc/oauth2_proxy/basic-auth-secret
              readOnly: true
            {{- end }}
```

I think `auth2_proxy.yaml` should be removed, and this yaml above from `deployment.yaml` just needs to be adjusted to use `basicAuth.secretName` and `basicAuth.filename` to mount the user's provided Secret into the path of `/etc/oauth2_proxy/auth_proxy_htpasswd`.

Also, you'll have to put this back if `basicAuth.enabled`:

```diff
-            - name: OAUTH2_PROXY_HTPASSWD_FILE
-              value: /etc/oauth2_proxy/auth_proxy_htpasswd
```